File: backend/src/models/student.ts
import { InferSchemaType, model, Schema } from "mongoose";

const studentSchema = new Schema({
    username: { type : String, required: true, unique: true },
    email: { type : String, required: true, unique: true },
    password: { type : String, required: true, select: false },
    registration_no: { type: String },
    name: { type: String },
    profile_image_url: {type:String},
    dob: { type: String },
    mobile: {type: String },
    year: {type:String },
    department: {type: String },
    division: {type: String },
    passing_year: {type: String },
    otp:{ type:String, default:'', select:false },
    present: { type: Boolean, default: false },
    events_registered: [
        {
          event: {
            type: Schema.Types.ObjectId,
            ref: "event",
            required: true,
          }
        },
      ],
      events_attended: [
        {
          event: {
            type: Schema.Types.ObjectId,
            ref: "event",
            required: true,
          }
        },
      ],
});
studentSchema.set("toJSON", {
    transform: function (doc, ret) {
      delete ret.events_registered;
      delete ret.events_attended;
      return ret;
    },
  });

type Student = InferSchemaType<typeof studentSchema>;

export default model<Student>("Student", studentSchema);

File: event-master/app/src/main/java/com/vinodpatildev/eventmaster/data/model/Student.kt
package com.vinodpatildev.eventmaster.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Student(
    @SerializedName("db_version")
    val __v: Int,
    @SerializedName("_id")
    val _id: String,
    @SerializedName("department")
    val department: String,
    @SerializedName("division")
    val division: String,
    @SerializedName("dob")
    val dob: String,
    @SerializedName("email")
    val email: String,
    @SerializedName("mobile")
    val mobile: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("passing_year")
    val passing_year: String,
    @SerializedName("password")
    val password: String,
    @SerializedName("registration_no")
    val registration_no: String,
    @SerializedName("username")
    val username: String,
    @SerializedName("year")
    val year: String
): Serializable {
    companion object {
        val TAG = "student"
    }
}